src/ol/renderer/canvas/canvasvectorlayerrenderer.js

```diff
@@ -194,6 +227,14 @@ ol.renderer.canvas.VectorLayer.prototype.prepareFrame =
 
   var extent = ol.extent.buffer(frameStateExtent,
       vectorLayerRenderBuffer * resolution);
+  var projectionExtent = viewState.projection.getExtent();
+
+  if (vectorSource.getWrapX() && viewState.projection.isGlobal() &&
```

Comment:
The `global` option on `ol.proj.Projection` and the `#isGlobal()` and `#setGlobal()` methods are marked stable. So we could only remove the `wrapX` option on all sources, which would mean that if you don't want a layer that could be wrapped around the x-axis to wrap, you would have to clone its projection and set `global` to `false`. Not so convenient IMHO.